src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: Function;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle' as const;
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (Math.min(a, b, c) < 0) {
+      throw new Error('your error message');
+    }
+
+    const [min, mid, max] = [a, b, c].sort((x, y) => x - y);
 
-export class Triangle implements Figure {}
+    if (max >= mid + min) {
+      throw new Error('your error message');
```

Comment:
[CODE STYLE] - The error message should provide a detailed description of the error. For example, 'The sum of any two sides of a triangle must be greater than the third side.'